operator/src/network/spec.rs

```diff
@@ -265,4 +265,56 @@ pub struct ResourceLimitsSpec {
 pub struct MonitoringSpec {
     /// Deploy monitoring resources into the network namespace directly
     pub namespaced: Option<bool>,
+    /// Deploy pod monitors
+    pub pod_monitoring: Option<PodMonitoringSpec>,
+}
+
+/// Describes the pod monitor configuration
+#[derive(Serialize, Deserialize, Debug, Default, PartialEq, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct PodMonitoringSpec {
+    /// Whether pod monitoring is enabled
+    pub enabled: Option<bool>,
```

Comment:
since the parent level is optional, is there a reason to also make this field optional?

Reply:
err.. maybe nevermind given these.. as this is what the user interacts with.

```rust
#[derive(Default)]
struct MonitoringConfig {
    namespaced: bool,
    pod_monitoring: PodMonitoringConfig,
}

#[derive(Default)]
struct PodMonitoringConfig {
    enabled: bool,
    pod_target_labels: Vec<String>,
}
```

Reply:
Yeah, I had the same question. We decided to do it this way to mimic the general pattern for how k8s resources are configured, i.e. they can be configured but not enabled.